### ISSUE TITLE
complex matmul and einsum

### ISSUE THREAD

**Issue by hknerdgn** [cupy/cupy#915]
I was wondering if complex matmul can be enabled. Actually, I need complex einsum and I found that it uses matmul and it does not currently work with complex numbers. I saw that some code is currently commented out in core.pyx which seems to be for complex arrays. I tried cupy.dot and it works with complex arrays. I checked numpy's einsum and it only uses numpy.dot not numpy.matmul, so dependence of einsum to matmul can be removed. I have not looked into this enough to be sure.

**Comment by hknerdgn** [comment]
I just removed those comments in core.pyx lines 3596 - 3613 and reinstalled cupy and  cupy.einsum and cupy.matmul seems to work with complex arrays now. I compared a single einsum example with random complex arrays in numpy and cupy and verified I get the same result which is nice. I saw that there is a test_einsum.py which probably needs to include all possible complex cases to test. Anyone more familiar with the testing can help test complex cases? Any reason not to include complex arrays in cupy.einsum?
